extensions/ql-vscode/src/databases/db-config-store.ts

```diff
@@ -0,0 +1,45 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { cloneDbConfig, DbConfig } from './db-config';
+
+export class DbConfigStore {
+  private readonly configPath: string;
+
+  private config: DbConfig;
+
+  public constructor(workspaceStoragePath: string) {
+    this.configPath = path.join(workspaceStoragePath, 'dbconfig.json');
+
+    this.config = emptyConfig;
+  }
+
+  public async initialize(): Promise<void> {
+    await this.loadConfig();
+  }
+
+  public getConfig(): DbConfig {
+    // Clone the config so that it's not modified outside of this class.
+    return cloneDbConfig(this.config);
+  }
+
+  private async loadConfig(): Promise<void> {
+    if (!await fs.pathExists(this.configPath)) {
+      await fs.createFile(this.configPath);
```

Comment:
Do we need this call to `createFile`? I would expect `writeJSON` to also create the file.

Reply:
Oh I didn't realise `writeJSON` creates the file too - that's handy. Removed `createFile`.